app/submitter/submitter.py

```diff
@@ -39,7 +40,14 @@ def send_message(self, message: str, tx_id: str, case_id: str) -> bool:
 
         # DEFAULT_RETRY is not idempotent.
         # However, this behaviour was deemed acceptable for our use case.
-        blob.upload_from_string(str(message).encode("utf8"), retry=DEFAULT_RETRY)
+        # Treats doubble submission error as successful because GCS doesn't allow partial data to be uploaded.
```

Comment:
This isn't strictly true. Maybe something like:
```
If an object exists then the GCS Client will attempt to delete the existing object before reuploading.
However, runner does not delete permission in an attempt to reduce duplicate receipts.
The first version of the object is acceptable as it is an extreme edge case for two submissions to contain different response data.
```

Also, I suggest adding this comment above the if statement instead of the entire block.